microcosm_pubsub/envelope.py

```diff
@@ -9,20 +9,106 @@
 from abc import ABCMeta, abstractmethod
 from json import loads
 from hashlib import md5
+from six import add_metaclass
 
 from microcosm.api import defaults
 
 from microcosm_pubsub.codecs import MediaTypeSchema, PubSubMessageCodec
 from microcosm_pubsub.message import SQSMessage
 
 
-class SQSEnvelope(object):
+@add_metaclass(ABCMeta)
```

Comment:
Big picture here:
 - We have two parsing steps, one that pulls out "our" parts of the message from SQS and one that translates this message into a media type and dictionary.
 - When working with `LocalStack` the first of these functions is abnormal.
 - We already had some support for generalizing the second of these parse functions
 - The changes here rewrite the SQS envelope parse into two abstract classes so we can mix and match